lib/ProductOpener/Display.pm

```diff
@@ -7133,14 +7163,13 @@ HTML
 	}
 
 	print header(%$http_headers_ref);
-	
-	my $status = $request_ref->{status};
-	if (defined $status) {
-		print header ( -status => $status );
-		my $r = Apache2::RequestUtil->request();
-		$r->rflush;
-		$r->status($status);
-	}
+
+	my $r = Apache2::RequestUtil->request();
+	$r->rflush;
+	# Setting the status makes mod_perl append a default error to the body
+	# $r->status($status);
+	# Send 200 instead.
+	$r->status(200);
```

Comment:
For me we should return Apache::Const::OK and propagate this response.

Or maybe in case of $status != 200, we should just use exit() but keep status in request.

Reply:
I tried returning Apache::Const::OK but it does not work, Apache still adds a default error body if $r's status is set to anything else than 200.

To test it, I wrote a simple test:

```
use ProductOpener::PerlStandards;

use CGI::Carp qw(fatalsToBrowser);

use Apache2::Const -compile => qw(OK);
use CGI qw/:cgi :form escapeHTML/;


my $status = param("status") || "200";

print header ( -status => $status );

my $body = "Status is set to $status";

binmode(STDOUT, ":encoding(UTF-8)");

print $body;

my $r = Apache2::RequestUtil->request();
$r->rflush;
$r->status($status);

return Apache2::Const::OK;
```

If I set the status to anything than 200, I get the default message added:

![image](https://user-images.githubusercontent.com/8158668/190980997-9c942f9a-c7e6-4197-9a58-85097323beec.png)

Reply:
ok